containers/Dockerfile.al2

```diff
@@ -1,90 +1,108 @@
+RUN cmake3 --build aws-c-auth/build --parallel $(nproc) --target install
 
 RUN git clone -b json-c-0.16-20220414 https://github.com/json-c/json-c.git
-RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SHARED_LIBS=OFF -GNinja -S json-c -B json-c/build
-RUN cmake3 --build json-c/build --target install
+RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DBUILD_TESTING=0 -DCMAKE_INSTALL_PREFIX=/usr -DBUILD_SHARED_LIBS=OFF -GNinja -S json-c -B json-c/build
+RUN cmake3 --build json-c/build --parallel $(nproc)  --target install
 
 RUN git clone -b v0.2.1 https://github.com/aws/aws-nitro-enclaves-nsm-api.git
-RUN source $HOME/.cargo/env && cd aws-nitro-enclaves-nsm-api && cargo build --release -p nsm-lib
+RUN source $HOME/.cargo/env && cd aws-nitro-enclaves-nsm-api && cargo build --release --jobs $(nproc) -p nsm-lib
 RUN mv aws-nitro-enclaves-nsm-api/target/release/libnsm.so /usr/lib64
 RUN mv aws-nitro-enclaves-nsm-api/target/release/nsm.h /usr/include
 
-RUN yum install -y doxygen
 COPY . aws-nitro-enclaves-sdk-c
 RUN cmake3 -DCMAKE_PREFIX_PATH=/usr -DCMAKE_INSTALL_PREFIX=/usr -GNinja \
 	-S aws-nitro-enclaves-sdk-c -B aws-nitro-enclaves-sdk-c/build
-RUN cmake3 --build aws-nitro-enclaves-sdk-c/build --target install
-RUN cmake3 --build aws-nitro-enclaves-sdk-c/build --target docs
-
-# Test
-FROM builder as test
-RUN cmake3 --build aws-nitro-enclaves-sdk-c/build --target test
+RUN cmake3 --build aws-nitro-enclaves-sdk-c/build --parallel $(nproc) --target install
+RUN cmake3 --build aws-nitro-enclaves-sdk-c/build --parallel $(nproc) --target docs
 
 # kmstool-enclave
-FROM amazonlinux as kmstool-enclave
+RUN mkdir -p /rootfs
+WORKDIR /rootfs
+
+RUN BINS="\
+    /usr/lib64/libnsm.so \
+    /usr/bin/kmstool_enclave \
+    " && \
+    for bin in $BINS; do \
+        { echo "$bin"; ldd "$bin" | grep -Eo "/.*lib.*/[^ ]+"; } | \
+            while read path; do \
+                mkdir -p ".$(dirname $path)"; \
+                cp -fL "$path" ".$path"; \
+            done \
+    done
+
+RUN mkdir -p /rootfs/etc/pki/tls/certs/ \
+    && cp -f /etc/pki/tls/certs/* /rootfs/etc/pki/tls/certs/
+RUN find /rootfs
+
+FROM scratch as kmstool-enclave
+
+COPY --from=builder /rootfs /
 
-# TODO: building packages statically instead of cleaning up unwanted packages from amazonlinux
-RUN rpm -e python python-libs python-urlgrabber python2-rpm pygpgme pyliblzma python-iniparse pyxattr python-pycurl amazon-linux-extras yum yum-metadata-parser yum-plugin-ovl yum-plugin-priorities
-COPY --from=builder /usr/lib64/libnsm.so /usr/lib64/libnsm.so
-COPY --from=builder /usr/bin/kmstool_enclave /kmstool_enclave
 ARG REGION
 ARG ENDPOINT
 ENV REGION=${REGION}
 ENV ENDPOINT=${ENDPOINT}
-CMD ["/kmstool_enclave"]
+CMD ["/usr/bin/kmstool_enclave"]
 
 # kmstool-instance
-FROM amazonlinux as kmstool-instance
+FROM amazonlinux:2 as kmstool-instance
 
 # TODO: building packages statically instead of cleaning up unwanted packages from amazonlinux
```

Comment:
These TODOs seem not very useful. I'd consider removing them.

Reply:
Only if we change things for the other binaries in another PR at some point.